### PR TITLE
fixes #10 so we can use CD to keep the fmp version up to date

### LINKED ISSUE #10: avoid hard coding the fabric8-maven-plugin and use a maven property instead
the fmp version is hard coded here:
https://github.com/obsidian-toaster/obsidian-addon/blob/master/src/main/resources/org/obsidiantoaster/generator/ui/starter/plugin-template.xml.ftl#L13

it'd be good to switch this with a maven property; we can then keep that version upgraded automatically as we cut new releases of fmp like we do for the rest of the fabric8 ecosystem

### PR DESCRIPTION
**Pull request by jstrachan**

### CODE REVIEW

**Comment by jstrachan** [comment]
@gastaldi does this look OK? I want to avoid using a really old version of fabric8-maven-plugin just for half of the obsidian wizard

**Comment by cmoulliard** [comment]
As you also change the version of the template, we must first discuss if we will align the quickstarts too to use this more recent version of f-m-p. Otherwise, we will include different versions of f-m-p and that will complicate the management of the releasing process

**Comment by jstrachan** [comment]
ah I didn't realise the quickstarts were so old; sorry I could have sworn I saw them using 3.2.x of fmp: https://github.com/obsidian-toaster-quickstarts/rest_springboot-tomcat/blob/master/pom.xml#L42

OK how about I change this PR to use the same fmp version; then you can manage all fmp versions in a simple way like we do with the fabric8 ecosystem https://github.com/fabric8io/fabric8-maven-dependencies/ - you could fork that repo and use it to maintain your global sets of versions of mvn plugins, boms & dependencies - you can then store the fmp version in one place & ensure consistent upgrades.

BTW the 3.1.x version is very old; I highly recommend upgrading ASAP

**Comment by cmoulliard** [comment]
We are still using the version 3.1.96 as the packaging war used for the Swarm quickstarts only work with that version & configuration [added](https://github.com/obsidian-toaster-quickstarts/rest_wildfly-swarm/blob/master/pom.xml#L144-L150) 

Otherwise, with 3.2, we get this error 

```
updated fabric8 plugin gives this error with WF Swarm: [ERROR] Failed to execute goal 
io.fabric8:fabric8-maven-plugin:3.2.26:build (default) on project wildfly-swarm-rest: Execution 
default of goal io.fabric8:fabric8-maven-plugin:3.2.26:build failed:

S2I not yet supported for the webapp-generator. Use -Dfabric8.mode=kubernetes or -
Dfabric8.buildStrategy=docker for OpenShift mode. Please refer to the reference manual at 
https://maven.fabric8.io  for details about build modes.
```

**Comment by jstrachan** [comment]
Ah OK thanks for the heads up. Looks like we need to fix that ASAP.

There's been so many fixes on the kubernetes/openshift YAML side of things am sure we'll hit issues with 3.1.x soon too.

**Comment by jstrachan** [comment]
@cmoulliard BTW here's a PR that fixes WAR packaging with spring boot / wildfly swarm apps in fabric8-maven-plugin 3.2.x

File: src/main/resources-filtered/org/obsidiantoaster/generator/ui/starter/plugin-template.xml.ftl [hunk 1]
```diff
@@ -10,7 +10,7 @@
          <plugin>
             <groupId>io.fabric8</groupId>
             <artifactId>fabric8-maven-plugin</artifactId>
-            <version>3.1.92</version>
+            <version>${version.fabric8-maven-plugin}</version>
```
Comment (gastaldi) [inline src/main/resources-filtered/org/obsidiantoaster/generator/ui/starter/plugin-template.xml.ftl]: This is a template and I think the property may not exist in the target.

Comment (jstrachan) [inline src/main/resources-filtered/org/obsidiantoaster/generator/ui/starter/plugin-template.xml.ftl]: the property is replaced during the build of the addon; so its never a maven property in the template thats actually used by the addon

Comment (jstrachan) [inline src/main/resources-filtered/org/obsidiantoaster/generator/ui/starter/plugin-template.xml.ftl]: i.e. at runtime the forge addon uses what gets generated into `target/classes` which has no maven properties any more

**Comment by gastaldi** [comment]
Got it. Wdyt @cmoulliard? Is this safe to merge?

**Comment by jstrachan** [comment]
@cmoulliard @gastaldi how about I modify the PR to keep the same fmp version for now - then we can upgrade fmp separately with a really trivial PR when folks are happy with the 3.2.x of fmp

**Comment by jstrachan** [comment]
OK I updated the PR to keep the fmp version being defined in the pom.xml via a maven property but reverted upgrading it for now - we can do that later

**Comment by gastaldi** [comment]
Thanks!

**Pull request merged**